sky/skylet/providers/gcp/config.py

```diff
@@ -773,28 +777,31 @@ def get_usable_vpc(config) -> str:
 
     If not found, create a new one with sufficient firewall rules.
 
+    Returns:
+        vpc_name: The name of the VPC network.
+        subnet_name: The name of the subnet in the VPC network for the specific
+            region.
+
     Raises:
       RuntimeError: if the user has specified a VPC name but the VPC is not found.
     """
     _, _, compute, _ = construct_clients_from_provider_config(config["provider"])
+    # For backward compatibility, reuse the VPC if the VM is launched.
```

Comment:
Good catch! Q: it seems like sky/provision/aws/config.py doesn't have such back-compat code?

Reply:
Good question! I just tried to create a new AWS VPC and after a cluster is provisioned, and run the `sky launch` again on that cluster. It seems `config.py` choose a different VPC, but the launching of that cluster still succeeded and the VPC of that cluster is still the old one. Probably, AWS's API ignores the different VPC for an existing VM.

Reply:
Got it. Maybe worth a test/comment here on whether GCP has the same property.

Reply:
That sounds good. 

Q: before step 3, is the single-node cluster stopped, or is it live? Wondering if the behavior applies to relaunching(starting) a stopped VM as wel.

Reply:
I tried both stopped and UP clusters, and it seems the steps above work correctly for both.